dnf/transaction_sr.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import libdnf
+
+from dnf.i18n import _
+import dnf.exceptions
+
+import json
+
+
+VERSION_MAJOR = "1"
+VERSION_MINOR = "0"
+VERSION = "%s.%s" % (VERSION_MAJOR, VERSION_MINOR)
```

Comment:
Why do you want to use a combination of major and minor version? Why to not use a sequential number?

Reply:
Because each of (MAJOR, MINOR) has a different semantics. Increasing MAJOR means you're breaking backwards compatibility (old dnf won't work with new transaction JSON). Increasing MINOR means you keep backwards compatibility (old dnf can work with new transaction JSON) but break forward compatibility (new dnf won't work with old transaction JSON).

IOW, when you're only adding stuff to the JSON, but don't change the old one, you increase MINOR. When you change what is already in there, you increase MAJOR.

I'll add this explanation as a docstring.